File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Particulate Matter Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var pm = new Tinkerforge.BrickletParticulateMatter(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current PM concentration
        pm.getPMConcentration(
            function (pm10, pm25, pm100) {
                console.log('PM 1.0: ' + pm10 + ' µg/m³');
                console.log('PM 2.5: ' + pm25 + ' µg/m³');
                console.log('PM 10.0: ' + pm100 + ' µg/m³');
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change XYZ to the UID of your Particulate Matter Bricklet

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_particulate_matter import BrickletParticulateMatter

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    pm = BrickletParticulateMatter(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get current PM concentration
    pm10, pm25, pm100 = pm.get_pm_concentration()

    print("PM 1.0: " + str(pm10) + " µg/m³")
    print("PM 2.5: " + str(pm25) + " µg/m³")
    print("PM 10.0: " + str(pm100) + " µg/m³")

    input("Press key to exit\n") # Use raw_input() in Python 2
    ipcon.disconnect()
